tests/typegen/strats/test_inline.py

```diff
@@ -0,0 +1,378 @@
+
+
+def test_callables():
+    resource_path = pathlib.Path("tests", "resource", "typegen", "callable.py")
+    assert resource_path.is_file()
+
+    c_clazz = "C"
+
+    traced = pd.DataFrame(columns=constants.TraceData.SCHEMA.keys())
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        None,
+        "add",
+        1,
+        TraceDataCategory.FUNCTION_ARGUMENT,
+        "x",
+        "int",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        None,
+        "add",
+        1,
+        TraceDataCategory.FUNCTION_ARGUMENT,
+        "y",
+        "int",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        None,
+        "add",
+        0,
+        TraceDataCategory.FUNCTION_RETURN,
+        "add",
+        "int",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "",
+        0,
+        TraceDataCategory.CLASS_MEMBER,
+        "a",
+        "int",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "",
+        0,
+        TraceDataCategory.CLASS_MEMBER,
+        "b",
+        "str",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "",
+        0,
+        TraceDataCategory.CLASS_MEMBER,
+        "c",
+        "bool",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "method",
+        8,
+        TraceDataCategory.FUNCTION_ARGUMENT,
+        "n",
+        "str",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "method",
+        8,
+        TraceDataCategory.FUNCTION_ARGUMENT,
+        "s",
+        "str",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "",
+        10,
+        TraceDataCategory.LOCAL_VARIABLE,
+        "d",
+        "float",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "",
+        11,
+        TraceDataCategory.LOCAL_VARIABLE,
+        "e",
+        "NoneType",
+    ]
+
+    traced.loc[len(traced.index)] = [
+        str(resource_path),
+        c_clazz,
+        "method",
+        0,
+        TraceDataCategory.FUNCTION_RETURN,
+        "method",
+        "bytes",
+    ]
+
+    gen = Generator(ident=InlineGenerator.ident, types=traced)
+    hinted = gen._gen_hints(
+        applicable=traced, nodes=ast.parse(source=resource_path.open().read())
+    )
+
+    result = ast.unparse(hinted)
+    logging.debug(f"\n{ast.unparse(hinted)}")
+
+    for node in ast.walk(hinted):
+        HintTest().visit(node)
+
+
+def test_assignments():
+    resource_path = pathlib.Path("tests", "resource", "typegen", "assignments.py")
+    assert resource_path.is_file()
+
+    gen = Generator(ident=InlineGenerator.ident, types=pd.DataFrame())
```

Comment:
Change to test_object and use the constructor of InlineGenerator directly.

Reply:
there is no constructor for InlineGenerator, and there needn't be one.
The factory method __new__ takes care of the set members